File: practice/Leetcode/1.两数之和/code.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        index = []
        i = 0
        j = len(nums) - 1
        cnums = sorted(nums.copy())
        result = cnums[i] + cnums[j]
        while result != target:
            if result > target:
                j -= 1
            else:
                i += 1
            result = cnums[i] + cnums[j]
        index.append(nums.index(cnums[i]))
        nums[index[0]] = cnums[j] + 1
        index.append(nums.index(cnums[j]))
        index = sorted(index)
        return index

File: practice/Leetcode/1.两数之和/code.cpp
class Solution {  
public:  
    vector<int> twoSum(vector<int>& data, int target) {  
        vector<int>cdata(data);//用于存储排序数据  
        vector<int>index;//索引  
        partial_sort_copy(data.begin(), data.end(), cdata.begin(), cdata.end());//拷贝排序  
        int i = 0, j = cdata.size() - 1;//双下标  
        int result = cdata[i] + cdata[j];  
        while(result != target){  
            if (result > target){  
                --j;//向前移动下标  
            }  
            else{  
                ++i;//向后移动下标  
            }  
            result = cdata[i] + cdata[j];  
        }  
        int pos = find(data.begin(), data.end(), cdata[i]) - data.begin();  
        index.push_back(pos);  
        data.at(pos) = cdata[j] + 1;//防止该位置被二次检索  
        index.push_back(find(data.begin(), data.end(), cdata[j]) - data.begin());  
        sort(index.begin(), index.end());  
        return index;  
    }  
};  